server/events.go

```diff
@@ -1067,23 +1066,32 @@ func (s *Server) sendLeafNodeConnectMsg(accName string) {
 // sendAccConnsUpdate is called to send out our information on the
 // account's local connections.
 // Lock should be held on entry.
-func (s *Server) sendAccConnsUpdate(a *Account, subj string) {
+func (s *Server) sendAccConnsUpdate(a *Account, subj ...string) {
 	if !s.eventsEnabled() || a == nil {
 		return
 	}
-	a.mu.RLock()
-
-	// Build event with account name and number of local clients and leafnodes.
-	m := AccountNumConns{
-		Account:    a.Name,
-		Conns:      a.numLocalConnections(),
-		LeafNodes:  a.numLocalLeafNodes(),
-		TotalConns: a.numLocalConnections() + a.numLocalLeafNodes(),
+	// Build event with account name and number of local clients and leafnodes. (if there is any interest in receiving)
+	var m *AccountNumConns
+	for _, sub := range subj {
+		if s.sys != nil {
+			if sac := s.sys.account; sac != nil && !sac.SubscriptionInterest(sub) {
```

Comment:
Careful: SubscriptionInterest(), although public, was really added for tests. For instance, it does not support wildcards (see comments on that function). That being said, as far as I can tell, you are calling this function only with literals, right?

Reply:
Other issue I see with suppressing interest, is that this is just looking for interest registered in the account. But we have situations (for instance case for gateways) where interest is not stored in an account, but instead of gw structures. Asking gw to send a message will then check if it has to be done or not (based on known no-interest, optimistic mode, etc..). 
So I am not sure, but you may be breaking things here?

Reply:
@kozlovic dang, I did not consider gateways. Basically I won't be able to use SubscriptionInterest or anything like it unless gateways will change. I'll remove the check for interest and basically publish the same message twice.

Reply:
Is that an issue to publish twice? (except for the "waste")

Reply:
I don't see one.